src/modules/AddOnOthers.rb

```diff
@@ -58,18 +58,65 @@ def Read
     #        </add_on_others>
     #      </add-on>
     def Export
-      others = @add_on_others.map do |p|
-        { "media_url"   => p["url"],
-          "alias"       => p["alias"],
-          "priority"    => p["priority"],
-          "name"        => p["name"],
-          "product_dir" => p["product_dir"] }
+      others = @add_on_others.each_with_object([]) do |addon, all|
+        next if registered_addon?(addon)
+
+        all << {
+          "media_url"   => addon["url"],
+          "alias"       => addon["alias"],
+          "priority"    => addon["priority"],
+          "name"        => addon["name"],
+          "product_dir" => addon["product_dir"]
+        }
       end
       { "add_on_others" => others }
     end
 
     publish function: :Export, type: "map ()"
     publish function: :Read, type: "map()"
+
+  private
+
+    # Determine whether an addon corresponds to a registered product
+    #
+    # @param addon [Hash] Addon data
+    # @return [Boolean]
+    def registered_addon?(addon)
+      return false unless addon["url"]
+
+      url = normalize_url(addon["url"])
+      registered_repositories_urls.include?(url)
+    end
+
+    # Returns the URLs corresponding to registered products repositories
+    #
+    # @return [Array<URI>]
+    def registered_repositories_urls
+      return @registered_repositories_urls if @registered_repositories_urls
+
+      begin
+        require "suse/connect"
```

Comment:
I'd rather avoid the direct usage of SUSEConnect. You can use [`registration/registration`](https://github.com/yast/yast-registration/blob/master/src/lib/registration/registration.rb) implemented in YaST here.
```suggestion
        require "registration/registration"
```

It provides the [`activated_products`](https://github.com/yast/yast-registration/blob/master/src/lib/registration/registration.rb#L170) method which you can use here.